client/src/partials/journal/journal.js

```diff
@@ -29,39 +29,49 @@ JournalController.$inject = [
  *
  * @module bhima/controllers/JournalController
  */
-function JournalController(Transactions, Sorting, Grouping, Pagination) {
+function JournalController(Transactions, Sorting, Grouping, Pagination, Filtering) {
   var vm = this;
 
-  // Journal utilities
-  var sorting, grouping, pagination;
+  // Journal utilites
+  var sorting, grouping, pagination, filtering;
 
   // gridOptions is bound to the UI Grid and used to configure many of the
   // options, it is also used by the grid to expose the API
   vm.gridOptions = {};
 
+  // Initialise each of the journal utilites, providing them access to the journal
+  // configuration options
+  sorting    = new Sorting(vm.gridOptions);
+  grouping   = new Grouping(vm.gridOptions);
+  pagination = new Pagination(vm.gridOptions, Transactions.list.data);
+  filtering  = new Filtering();
+
   // bind the transactions service to populate the grid component
   vm.gridOptions.data = Transactions.list.data;
 
-  // Initialise each of the journal utilities, providing them access to the journal
-  // configuration options
-  sorting = new Sorting(vm.gridOptions);
-  pagination = new Pagination(vm.gridOptions, Transactions.list.data);
-  grouping = new Grouping(vm.gridOptions);
+  // enable filter option
+  vm.gridOptions.enableFiltering = true;
```

Comment:
All other journal services import the `gridOptions` to set up specific options.  For example, [this line](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/js/services/journal/JournalPagination.js#L97) in the pagination services configures external pagination.  In the same way, sorting sets up sorting in [this line](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/js/services/journal/JournalSorting.js#L91).

This is a nice pattern developed by @sfount.  I propose that you refactor filtering to take in `gridOptions` like this:
```js
pagination = new Pagination(vm.gridOptions, Transactions.list.data);
filtering  = new Filtering(vm.gridOptions);
```
and set up the `enableFiltering` property there.